### PR TITLE
Add FXIOS-9152 Display search provider icon in Client

### PR DESCRIPTION
**Pull request by thatswinnie**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-9152)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/20275)

## :bulb: Description
Displays the search engine icon in the address toolbar. Also fixes a few UI bugs for the location view and simplifies the code.

Simulator Screenshot - iPhone 15 Pro Max - 2024-05-31 at 16 52 37

Simulator Screenshot - iPhone 15 Pro Max - 2024-05-31 at 17 12 23

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [x] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: SampleBrowser/SampleBrowser/Model/RootViewControllerModel.swift [hunk 1]
```diff
@@ -88,14 +89,14 @@ class RootViewControllerModel {
...
-            clearButtonA11yLabel: "Clean",
             searchEngineImageViewA11yId: "searchEngine",
             searchEngineImageViewA11yLabel: "Search engine icon",
+            lockIconButtonA11yId: "lockButton",
+            lockIconButtonA11yLabel: "Tracking Protection",
             urlTextFieldPlaceholder: "Search or enter address",
             urlTextFieldA11yId: "urlTextField",
             urlTextFieldA11yLabel: "Address Bar",
-            searchEngineImageName: "bingSearchEngine",
+            searchEngineImage: UIImage(named: "bingSearchEngine"),
```
[the remark this answers is not in the document]
Comment (thatswinnie) [reply SampleBrowser/SampleBrowser/Model/RootViewControllerModel.swift:99]: I'd love to do that but that's not how our search engines work. The icons for the search engine are saved base64 encoded in config files. If we want to pass the image name we'd have to save them to disk as an image first.

[the remark this answers is not in the document]
Comment (PARAIPAN9) [reply SampleBrowser/SampleBrowser/Model/RootViewControllerModel.swift:99]: I see now that this is in SampleBrowser, nevermind...

**Review by PARAIPAN9 — APPROVED** [review]

File: BrowserKit/Sources/ToolbarKit/LocationView/LocationTextField.swift [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - View setup
+    override func layoutSubviews() {
+        super.layoutSubviews()
+
+        guard let image = UIImage(named: StandardImageIdentifiers.Large.crossCircleFill) else { return }
+        if tintedClearImage == nil {
+            if let clearButtonTintColor = clearButtonTintColor {
```
Comment (cyndichin) [inline BrowserKit/Sources/ToolbarKit/LocationView/LocationTextField.swift:28]: nit
```suggestion
            if let clearButtonTintColor {
```

**Review by cyndichin — APPROVED** [review]
looks good, discussed live, seems we copied over the clear button logic from existing code from `ToolbarTextField`

**Pull request merged**